bitcoin/src/address.rs

```diff
@@ -1011,6 +1039,43 @@ impl Address {
     pub fn matches_script_pubkey(&self, script_pubkey: &Script) -> bool {
         self.payload().matches_script_pubkey(script_pubkey)
     }
+
+    /// Deserialize an address using serde and performing network check.
+    ///
+    /// This can be used in [`DeserializeSeed`](serde::de::DeserializeSeed) implementations to
+    /// deserialize an address with network check. Apart from being shorter than rolling your own, this
+    /// also provides much nicer error messages.
+    #[cfg(feature = "serde")]
+    pub fn deserialize_checked<'de, D>(
```

Comment:
Like in the `amount` module, the naming convention is `pub mod serde` and then either submodules in there to use with `#[serde(with = ` or functions in there to use with `#[serde(deserialize_with = `.

That way also the other types you created can be in this serde module.

Reply:
And I think this seed thing isn't super useful for people in practice. I think it would be more useful to have per-network deserializers.

Reply:
This is impossible since the function needs another parameter and `serde` doesn't have direct support for it.
If by "per-network deserializers" you mean functions then that might tempt people to just hardcode the network which is a terrible practice.